test/distributed/fsdp/test_fsdp_summon_full_params.py

```diff
@@ -38,9 +38,11 @@
 
 
 def _run_test_summon_full_param_writeback(cls, writeback, cpu_offload, modify_outer):
-    model = FSDP(
-        nn.Sequential(FSDP(nn.Linear(5, 5, bias=False)), nn.Linear(5, 3, bias=False))
-    ).cuda(cls.rank)
+    lin1 = FSDP(nn.Linear(5, 5, bias=False).cuda(cls.rank), cpu_offload=cpu_offload)
+    lin2 = nn.Linear(5, 3, bias=False).cuda(cls.rank)
+    model = FSDP(nn.Sequential(lin1, lin2), cpu_offload=cpu_offload)
```

Comment:
curious, what's the expectation on the input model when `cpu_offload==True`? Does it have to reside on GPU? Or is it OK to pass in a CPU model?

Reply:
Passing CPU offload with CPU model should work, as we don't assume any device in the constructor and would restore to GPU in forward pass before all_gather. Although, it does not seem that we have a test for this, we should add one.